client/src/App.tsx

```diff
@@ -1,38 +1,195 @@
+      };
+    });
+  };
+
+  const nextPlayer = () => {
+    const prevPlayer = currentPlayerRef.current;
+    const prevPlayerIndex = players.findIndex(
+      (player) => player.id === prevPlayer.id
+    );
+    let nextPlayer: Player;
+    if (prevPlayerIndex === players.length - 1) {
+      nextPlayer = players[0];
+    } else {
+      nextPlayer = players[prevPlayerIndex + 1];
+    }
+    currentPlayerRef.current = nextPlayer;
+    startRound(nextPlayer);
+  };
+
+  const startRound = (player: Player) => {
+    currentRoundRef.current = currentRoundRef.current + 1;
+
+    setGame((prevGame: Game) => {
+      return {
+        ...prevGame,
+        currentRound: prevGame.currentRound + 1,
+      };
+    });
+
+    showPlayerCard(player);
+
+    if (!player.isHuman) {
+      const stat = chooseRandomStat(
+        player.cards[0].stats,
+        player.cards[0].stats.length
+      );
+      setTimeout(() => {
+        playRound(stat, player);
+      }, DEFAULT_TIMEOUT);
+    }
+  };
+
+  const playRound = (stat: Stat, player: Player) => {
+    showAllCards();
+
+    setTimeout(() => {
+      const { id } = calculateRoundWinner(players, stat, player);
+      console.log("winner " + id);
+
+      setGame((prevGame: Game) => {
+        return {
+          ...prevGame,
+          players: updatePlayerScores(prevGame.players, id),
+          roundWinners: updateRoundWinners(prevGame.roundWinners, id),
+          //roundsPlayed: prevGame.roundsPlayed + 1,
+        };
+      });
+
+      endRound();
+    }, DEFAULT_TIMEOUT);
+  };
+
+  const endRound = () => {
+    hideAllCards();
+
+    setTimeout(() => {
+      setGame((prevGame: Game) => {
+        return { ...prevGame, players: updatePlayerCards(prevGame.players) };
+      });
+      if (currentRoundRef.current === totalRounds) {
+        console.log("end of game");
+      } else {
+        nextPlayer();
+      }
+    }, DEFAULT_TIMEOUT);
+  };
+
   return (
     <>
+      <Header />
       <PlayerScore
         playerName={player1.name}
         playerId={player1.id}
         updateName={() => {}}
         playerScore={player1.score}
-        currentRound={game.currentRound}
-        totalRounds={game.totalRounds}
-        roundWinners={[]}
+        currentRound={currentRoundRef.current}
+        totalRounds={totalRounds}
+        roundWinners={roundWinners}
       />
       <PlayerScore
         playerName={player2.name}
         playerId={player2.id}
         updateName={() => {}}
         playerScore={player2.score}
-        currentRound={game.currentRound}
-        totalRounds={game.totalRounds}
-        roundWinners={[]}
+        currentRound={currentRoundRef.current}
+        totalRounds={totalRounds}
+        roundWinners={roundWinners}
       />
-      <Header />
+      <button onClick={() => startRound(currentPlayerRef.current)}>
+        Start Round
+      </button>
+      <p>Current round is: {game.currentRound}</p>
```

Comment:
here we could use game.roundWinners.length

Reply:
See comment above for why roundWinners.length doesn't equate to currentRound